controllers/twophase/types.go

```diff
@@ -111,17 +109,27 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		// Start recover
 		r.Log.Info("Recovering")
 
-		err = r.Recover(ctx, req, chaos)
-		if err != nil {
-			r.Log.Error(err, "failed to recover chaos")
-			return ctrl.Result{Requeue: true}, err
+		// Don't need to recover again if chaos was paused before
+		if status.Experiment.Phase != v1alpha1.ExperimentPhasePaused {
+			if err = r.Recover(ctx, req, chaos); err != nil {
+				r.Log.Error(err, "failed to recover chaos")
+				return ctrl.Result{Requeue: true}, err
+			}
 		}
+
 		chaos.SetNextRecover(time.Time{})
 
 		status.Experiment.EndTime = &metav1.Time{
 			Time: time.Now(),
 		}
 		status.Experiment.Phase = v1alpha1.ExperimentPhaseFinished
+	} else if status.Experiment.Phase == v1alpha1.ExperimentPhasePaused &&
+		!chaos.GetNextRecover().IsZero() && chaos.GetNextRecover().After(now) {
+		// Resume
+		r.Log.Info("Resuming")
+		if err := applyAction(ctx, r, req, chaos); err != nil {
```

Comment:
I don't understand this logic. The phase is `Paused` and why needs to apply here?

Reply:
This is the time that we want to resume the chaos, which means changing the `Paused` status to `Running` again, so we need to apply the action and update phase to `Running`. Does this make sense to you?